pkg/envoy/ads/response_benchmark_test.go

```diff
@@ -0,0 +1,170 @@
+package ads
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	mapset "github.com/deckarep/golang-set"
+	"github.com/google/uuid"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sClientFake "k8s.io/client-go/kubernetes/fake"
+
+	configv1alpha2 "github.com/openservicemesh/osm/pkg/apis/config/v1alpha2"
+	"github.com/openservicemesh/osm/pkg/catalog"
+	"github.com/openservicemesh/osm/pkg/endpoint"
+	"github.com/openservicemesh/osm/pkg/k8s/informers"
+	"github.com/openservicemesh/osm/pkg/messaging"
+	"github.com/openservicemesh/osm/pkg/policy"
+	"github.com/openservicemesh/osm/pkg/providers/kube"
+	"github.com/openservicemesh/osm/pkg/signals"
+	"github.com/openservicemesh/osm/pkg/smi"
+
+	"github.com/openservicemesh/osm/pkg/certificate"
+
+	"github.com/openservicemesh/osm/pkg/configurator"
+	"github.com/openservicemesh/osm/pkg/constants"
+	"github.com/openservicemesh/osm/pkg/envoy"
+	"github.com/openservicemesh/osm/pkg/envoy/registry"
+	configFake "github.com/openservicemesh/osm/pkg/gen/client/config/clientset/versioned/fake"
+	policyFake "github.com/openservicemesh/osm/pkg/gen/client/policy/clientset/versioned/fake"
+	"github.com/openservicemesh/osm/pkg/k8s"
+	"github.com/openservicemesh/osm/pkg/logger"
+	"github.com/openservicemesh/osm/pkg/service"
+	"github.com/openservicemesh/osm/pkg/tests"
+)
+
+func BenchmarkSendXDSResponse(b *testing.B) {
```

Comment:
You're actually only testing send response for SDS here due to the `SetSubscribedResources` call.

Can we rename this so that it's clear this is only SDS, or you can add a loop over XDSResponseOrder to respond to all?

Alternatively we could have a single test for each so we know the individual benchmarks. Then with the comment below about separating out the server creation the tests themselves are much smaller

Reply:
Added more xDS server types to be benchmarked individually.